mds/management/commands/poll_providers.py

```diff
@@ -228,14 +229,21 @@ def prepare_status_changes(self, status_changes, provider):
                 event_type_reason = status_change["event_type_reason"]
             except KeyError:  # Spec violation!
                 self.stderr.write(
-                    "Warning: device %s has no event_type_reason" % device_id
+                    "Warning: device %s has no event_type_reason, ignoring." % device_id
                 )
-                # Let recording the device fail with a null value, but only that one in the batch
-                agency_event_type = None
-            else:
+                # Ignore just that status change to avoid rejecting the whole batch
+                continue
+            try:
                 agency_event_type = enums.PROVIDER_EVENT_TYPE_REASON_TO_AGENCY_EVENT_TYPE[
                     event_type_reason
                 ]
+            except KeyError:  # Spec violation!
+                self.stderr.write(
+                    'Warning: device %s has unknown "%s" event_type_reason, ignoring.'
+                    % (device_id, event_type_reason)
+                )
```

Comment:
use `self.log` to log exception and get the traceback on the sentry side.

Reply:
This is vanilla Django, we don't have `EnhancedCommandMixin`, but I still can reuse the logger I created.